components/style_traits/values.rs

```diff
@@ -72,11 +72,58 @@ where
     T: ToCss,
 {
     #[inline]
-    fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: Write {
+    fn to_css<W>(&self, dest: &mut CssWriter<W>) -> fmt::Result where W: Write {
         self.as_ref().map_or(Ok(()), |value| value.to_css(dest))
     }
 }
 
+/// A writer tailored for serialising CSS.
+///
+/// Coupled with SequenceWriter, this allows callers to transparently handle
+/// things like comma-separated values etc.
```

Comment:
Maybe put an example of how this is used? I had to read it a bit carefully.

In particular, what state does `prefix` represent? If I'm not wrong, when writing a sequence:

 * We start with prefix = `Some("")`.
 * On the first `item`, assuming it writes something, it `take`s the prefix.
 * On the second one, since there's no prefix, we set `prefix: Some(sequence_separator)`, and reset it after the `to_css` call begins.
 * ...